mobile/src/main/java/ch/epfl/sdp/mobile/ui/setting/EditProfileImageDialog.kt

```diff
@@ -0,0 +1,154 @@
+          items(items = Emojis) { item ->
+            SelectEmojiItem(selected = state.emoji == item, onClick = { state.emoji = item }, item)
+          }
+        }
+        LazyRow(contentPadding = PaddingValues(horizontal = 16.dp)) {
+          items(items = Color.values) { item ->
+            SelectBackgroundColorItem(
+                selected = state.backgroundColor == item,
+                onClick = { state.backgroundColor = item },
+                item)
+          }
+        }
+        Spacer(Modifier.height(16.dp))
+      }
+    }
+  }
+}
+
+/**
+ * Composes the edit settings picture which shows the edit changes
+ * @param backgroundColor background color for setting image
+ * @param emoji the emoji string
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+private fun EditSettingPicture(
+    backgroundColor: Color,
+    emoji: String,
+    modifier: Modifier = Modifier
+) {
+
+  Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = modifier.fillMaxWidth()) {
+    Box(
+        modifier = Modifier.size(118.dp).background(backgroundColor.toColor(), CircleShape),
+        contentAlignment = Alignment.Center,
+    ) { Text(emoji, style = MaterialTheme.typography.h3) }
+  }
+}
+
+/**
+ * Select Emoji Item
+ * @param selected boolean which check if current element is selected
+ * @param onCLick function to execute if current element is clicked
+ * @param emoji String which to be changed
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+private fun SelectEmojiItem(
+    selected: Boolean,
+    onClick: () -> Unit,
+    emoji: String,
+    modifier: Modifier = Modifier
+) {
+  Box(
+      modifier =
+          if (selected) modifier.size(72.dp).border(BorderStroke(4.dp, Green800), CircleShape)
+          else modifier.size(72.dp).clickable { onClick() },
+      contentAlignment = Alignment.Center,
+  ) { Text(emoji, style = MaterialTheme.typography.h4) }
+}
+
+/**
+ * Select Background color of profile Image
+ * @param selected boolean if current element is equal current item
+ * @param onClick function to execute if element is clicked
+ * @param backgroundColor the current color in item
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+private fun SelectBackgroundColorItem(
+    selected: Boolean,
+    onClick: () -> Unit,
+    backgroundColor: Color,
+    modifier: Modifier = Modifier
+) {
+  Box(modifier = modifier.padding(4.dp)) {
+    Box(
+        modifier =
+            if (selected)
+                modifier
+                    .size(64.dp)
+                    .background(backgroundColor.toColor(), CircleShape)
+                    .border(BorderStroke(4.dp, Green800), CircleShape)
+            else
+                modifier.size(64.dp).background(backgroundColor.toColor(), CircleShape).clickable {
+                  onClick()
+                },
+        contentAlignment = Alignment.Center,
+    ) {}
+  }
```

Comment:
Bug : Something is wrong with the Modifiers here, since you're using the `modifier` argument in one of the nested composables.

Additionally, I think you could use the Box overload with no content, and you should probably not nest boxes since you could just move the modifier.padding around.